docs/architecture/adr-template.md

```diff
@@ -8,16 +8,50 @@
 
 > This section contains all the context one needs to understand the current state, and why there is a problem. It should be as succinct as possible and introduce the high level idea behind the solution.
 
+## Alternative Approaches
+
+> This section contains information around alternative options that are considered before making a decision. It should contain a explanation on why the alternative approach(es) were not chosen.
+
 ## Decision
 
-> This section explains all of the details of the proposed solution, including implementation details.
-> It should also describe affects / corollary items that may need to be changed as a part of this.
-> If the proposed change will be large, please also indicate a way to do the change to maximize ease of review.
-> (e.g. the optimal split of things to do between separate PR's)
+> This section records the decision that was made.
+> It is best to record as much info as possible from the discussion that happened. This aids in not having to go back to the Pull Request to get the needed information.
+
+## Detailed Design
+
+> This section does not need to be filled in at the start of the ADR, but must be completed prior to the merging of the implementation.
+
+Here are some common questions that get answered as part of the detailed design:
+
+- What are the user requirements?
+
+- What systems will be affected?
+
+- What new data structures are needed, what data structures will be changed?
+
+- What new APIs will be needed, what APIs will be changed?
+
+- What are the efficiency considerations (time/space)?
+
+- What are the expected access patterns (load/throughput)?
+
+- Are there any logging, monitoring or observability needs?
+
+- Are there any security considerations?
+
+- Are there any privacy considerations?
+
+- How will the changes be tested?
+
+- If the change is large, how will the changes be broken up for ease of review?
+
+- Will these changes require a breaking (major) release?
+
+- Does this change require coordination with the SDK or other?
```

Comment:
Maybe this bit should be quoted with `>` like the rest of the template?

Reply:
You probably want to quote the empty lines as well, such that the whole section gets the blockquote marker instead of each individual line having its own blockquote marker.

Reply:
Added. wasn't sure if it should be all or only the questions. I don't have a preference so went with your suggestion.

Reply:
Thanks! Did you look at it rendered? It's pretty ugly and weird with separate blockquotes. :)

Reply:
yea, it looks a bit odd but I imagined someone either deleting it all or answering below the question so the space was there for that purpose. But the Author can decide and modify as they please

Reply:
Ah. I imagined this was more of a "did you remember" checklist, not a template for the actual content. Q&A bullet lists read a lot worse than actual prose. :)